### DISCUSSION TITLE
re-export minimal necessary glib call to simplify libvips foreign calls

### DISCUSSION THREAD

**Discussion by UrielCh** [libvips/libvips#3298] [General]
Hi,

I want to use [libvips](https://github.com/libvips/libvips) in a Deno script, I tried to use [sharp](https://github.com/lovell/sharp), but le lib looks to be incompatible.

So I tried to use it directly from deno using the FFI feature, and it works.

But I have a minor issue; in standard usage, the object needs to be released with gLib `g_object_unref`; in my case, calling `g_object_unref` requires me to re-map the glib (for a single call or maybe 2).

If there is a small number of gLib calls needed to use the API, can you re-export them from the main lib?

I have started le lib mapping [here](https://github.com/UrielCh/libvips-4-deno)

**Comment by lovell** [comment]
As an aside, the latest Deno now supports Node-API and therefore sharp - please see https://github.com/lovell/sharp/issues/2583#issuecomment-1310628079

**Reply by UrielCh** [reply]
I tried it yesterday. I will retry. (I prefer using sharp), let me commit my sample.

**Reply by UrielCh** [reply]
or not, that a sharp sample, I failed to make it works with deno.
```typescript
// @deno-types="npm:@types/sharp"
import sharp from 'sharp';
sharp('./img.jpg').resize(128).toFile('./img128.jpg');
```

**Comment by UrielCh** [comment]
just try it now:

```
error: Uncaught Error: 
Something went wrong installing the "sharp" module

Cannot find module '../build/Release/sharp-win32-x64.node'
Require stack:
- C:\Users\uriel\AppData\Local\deno\npm\registry.npmjs.org\sharp\0.31.3\lib\sharp.js
- C:\Users\uriel\AppData\Local\deno\npm\registry.npmjs.org\sharp\0.31.3\lib\constructor.js
- C:\Users\uriel\AppData\Local\deno\npm\registry.npmjs.org\sharp\0.31.3\lib\index.js
- C:\Users\uriel\AppData\Local\deno\npm\registry.npmjs.org\sharp\0.31.3\lib\index.js

Possible solutions:
- Install with verbose logging and look for errors: "npm install --ignore-scripts=false --foreground-scripts --verbose sharp"
- Install for the current win32-x64 runtime: "npm install --platform=win32 --arch=x64 sharp"
- Consult the installation documentation: https://sharp.pixelplumbing.com/install
    at Object.<anonymous> (file:///C:/Users/uriel/AppData/Local/deno/npm/registry.npmjs.org/sharp/0.31.3/lib/sharp.js:34:9)
    at Object.<anonymous> (file:///C:/Users/uriel/AppData/Local/deno/npm/registry.npmjs.org/sharp/0.31.3/lib/sharp.js:37:4)
    at Module._compile (deno:ext/node/02_require.js:747:36)
    at Object.Module._extensions..js (deno:ext/node/02_require.js:780:12)
    at Module.load (deno:ext/node/02_require.js:658:34)
    at Function.Module._load (deno:ext/node/02_require.js:515:14)
    at Module.require (deno:ext/node/02_require.js:680:21)
    at require (deno:ext/node/02_require.js:820:18)
    at Object.<anonymous> (file:///C:/Users/uriel/AppData/Local/deno/npm/registry.npmjs.org/sharp/0.31.3/lib/constructor.js:8:1)
    at Object.<anonymous> (file:///C:/Users/uriel/AppData/Local/deno/npm/registry.npmjs.org/sharp/0.31.3/lib/constructor.js:431:4)
```

**Reply by lovell** [reply]
Please ensure you carefully follow the steps in https://github.com/lovell/sharp/issues/2583#issuecomment-1310628079 ensuring any filesystem paths and version numbers are adjusted for your specific scenario e.g. `C:/Users/uriel/AppData/Local/deno/npm/registry.npmjs.org/sharp/0.31.3` rather than `~/.cache/deno/npm/registry.npmjs.org/sharp/0.31.2`.

This is off-topic for libvips, so please open a new issue against sharp if you still require help.
